Decaffeinate

WardCunningham opened the pull request:
This merges mods from last summer in support of custom registration workflows in use by wiki.cafe since then.
This also applies decaffeinate done manually for client and test and then mechanically for the server side.

WardCunningham committed:
new endpoints for custom registration

WardCunningham committed:
try importing register.js module

WardCunningham committed:
call with callback

WardCunningham committed:
delegate get to register.js

WardCunningham committed:
add argv to custom api

WardCunningham committed:
require owner for get or post

Bortseb committed:
add query param "domain" on custom get endpoint

also remove the need to be owner for just the get endpoint

WardCunningham committed:
Merge pull request #11 from Bortseb/patch-1

add query param "domain" on custom get endpoint

Bortseb committed:
Send whole request, not just optional param to adapter

WardCunningham committed:
Merge pull request #12 from Bortseb/ward/custom

Send whole request, not just optional param to adapter

WardCunningham committed:
publish as 0.5.0-pre6

WardCunningham committed:
merge conflic on version

WardCunningham committed:
rename to js

WardCunningham committed:
decaffeinate register

WardCunningham committed:
decaff test.coffee

WardCunningham committed:
decaf server, test

paul90 committed:
update git/npm  ignore, and update authors.

paul90 commented:
the force push as to step back to the previous commit and ensure that the wrong `server.js` was not committed.

the decaffeinated server code is currently missing from this PR.

WardCunningham committed:
add omitted script

WardCunningham committed:
add missing server.js

paul90 commented on server/server.js:
```diff
@@ -0,0 +1,240 @@
+    const like = looking.pop();
+    const where = looking.join('/');
+    return fs.readFile(`${argv.status}/owner.json`, 'utf8', function(err, owner) {
+      if (err) { return e500(`${err}`); }
```
getting an error that `e500` is not defined.

paul90 commented on server/server.js:
```diff
@@ -0,0 +1,240 @@
+      });
+    });
+  });
+});
```
premature end of startServer?

paul90 commented on server/server.js:
```diff
@@ -0,0 +1,240 @@
+      });
+    });
+  });
+});
```
premature end of startServer?

paul90 reviewed (commented):
a few initial observations.

the `})`s might look off because prettier hasn't been run!

WardCunningham committed:
finish promise sequence

paul90 committed:
the last bits missing from the template

paul90 committed:
prettier code

WardCunningham committed:
prettier server.js and test.js

WardCunningham committed:
fix merge conflicts

Bortseb committed:
Convert custom module import from coffeescript to JS

Bortseb committed:
prettier changes

WardCunningham committed:
Merge pull request #18 from Bortseb/decaffeinate

Convert custom register adapter module import from coffeescript to JS

Thank you. Will publish this as next pre-release.

WardCunningham committed:
0.5.0-pre8

Bortseb committed:
enable non-owners to post to custom endpoint, and send whole request to adapter

WardCunningham committed:
Merge pull request #19 from Bortseb/decaffeinate

enable non-owners to post to custom endpoint, and send whole request to adapter

Reviewed with Robert online.

WardCunningham committed:
0.5.0-pre9

Bortseb committed:
Update server.js

Improve the error message returned when a custom register adapter isn't found for the site.

paul90 committed:
Merge pull request #20 from Bortseb/decaffeinate

Update server.js
